src/main/java/dev/meyi/bn/config/Configuration.java

```diff
@@ -1,127 +1,118 @@
-  private boolean showSellOfferProfit;
-  private boolean showProfitPerMil;
-
-  public Configuration(boolean collectionCheck, int craftingSortingOption,
-      int craftingListLength, boolean suggestionShowEnchantments,
-      boolean showInstantSellProfit, boolean showSellOfferProfit, boolean showProfitPerMil,
-      int suggestionListLength, boolean showChatMessages, String apiKey, boolean useBuyOrders, double bazaarProfit, boolean bankRawDifference,
-      ModuleConfig[] modules) {
-    this.collectionCheck = collectionCheck;
-    this.craftingSortingOption = craftingSortingOption;
-    this.craftingListLength = craftingListLength;
-    this.suggestionShowEnchantments = suggestionShowEnchantments;
-    this.showInstantSellProfit = showInstantSellProfit;
-    this.showSellOfferProfit = showSellOfferProfit;
-    this.showProfitPerMil = showProfitPerMil;
-    this.suggestionListLength = suggestionListLength;
-    this.bazaarProfit = bazaarProfit;
-    this.bankRawDifference = bankRawDifference;
-    this.api =
-        apiKey == null ? "" : apiKey; // It is fixed in createDefaultConfig, but redundancies.
-    this.version = BazaarNotifier.VERSION;
-    this.modules = modules;
-    this.showChatMessages = showChatMessages;
-    this.useBuyOrders = useBuyOrders;
-  }
-
-
-  public static void saveConfig(File file, Configuration config) {
-    Gson gson = new Gson();
-    BazaarNotifier.config.modules = BazaarNotifier.modules.generateConfig();
-    BazaarNotifier.config.bazaarProfit = BankCalculator.getBazaarProfit();
-    try {
-      if (!file.isFile()) {
-        //noinspection ResultOfMethodCallIgnored
-        file.createNewFile();
-      }
-      Files.write(Paths.get(file.getAbsolutePath()),
-          gson.toJson(config).getBytes(StandardCharsets.UTF_8));
-    } catch (IOException e) {
-      e.printStackTrace();
-    }
-  }
-
-  public static Configuration createDefaultConfig() {
-    ModuleConfig[] c = new ModuleConfig[MODULE_LENGTH];
-    int i = 0;
-    for (ModuleName moduleName : ModuleName.values()) {
-      c[i++] = ModuleConfig.generateDefaultConfig(moduleName.name());
-    }
-    return new Configuration(Defaults.COLLECTION_CHECKING,
-        Defaults.CRAFTING_SORTING_OPTION, Defaults.CRAFTING_LIST_LENGTH,
-        Defaults.SUGGESTION_SHOW_ENCHANTMENTS,
-        Defaults.INSTANT_SELL_PROFIT, Defaults.SELL_OFFER_PROFIT,
-        Defaults.PROFIT_PER_MIL, Defaults.SUGGESTION_LIST_LENGTH, Defaults.SEND_CHAT_MESSAGES, "",
-        Defaults.USE_BUY_ORDERS, 0, Defaults.BANK_RAW_DIFFERENCE, c);
-  }
-
-  public boolean isShowSellOfferProfit() {
-    return showSellOfferProfit;
+  public Configuration() {
+    super(new Mod("BazaarNotifier", ModType.SKYBLOCK,"/icon.png", new JsonMigrator("./config/BazaarNotifier/config.json")), "bazaarnotifier.json");
+    initialize();
+    addDependency("collectionCheck", "Requires ApiKey", () -> {
+      if ("".equals(api)) collectionCheck = false;
+      return !"".equals(api);
```

Comment:
any reason not to just do `api.isEmpty()`?

or can api be null for some reason?